website/docs/tutorials/working-with-files.md

```diff
@@ -4,8 +4,127 @@ title: Working with Files
 sidebar_label: Working with Files
 ---
 
+Even if the core Solid data model is **structured** data (see [Working with data](./working-with-data)), a Solid Pod
+can also act as a regular general-purpose data store. Besides your profile document, your friend list and the likes, your
+Pod can also store your photos, PDFs and any other type of file. Note that the [access restrictions](./managing-access)
+apply to files the same way they apply to any other resource.
+
+Like anything else in your Pod, each file is a resource with a distinct URL, which may or may not contain a hint
+such as a `.jpg` extension for a photo. You'll find that the functions available to read and write files to/from
+your Pod are very close to the browser `fetch`: files are represented as typed [`Blob`s](https://developer.mozilla.org/en-US/docs/Web/API/Blob), and the result of the functions is returned as a [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response).
+
 :::caution
 
-This section is yet to be written.
+The functions described in this documentation are still early stage, and subject to breaking change.
 
 :::
+
+## Reading a file
+
+Reading a file is just a matter of fetching the content available at a certain URL. You get a `Response` in return, which contains
+the fetched file as a blob. It is then up to you to decode it appropriately.
+
+```typescript
+import { unstable_fetchFile } from "lit-solid";
+
+const response = await unstable_fetchFile(
+  "https://example.com/some/interesting/file"
+);
+if (response.ok) {
+  const myFile = await response.blob();
+}
+```
+
+## Deleting a file
+
+Deleting a file is also a simple operation: you just erase the content available at a certain URL.
+
+```typescript
+import { unstable_fetchFile } from "lit-solid";
+
+const response = await unstable_deleteFile(
+  "https://example.com/some/boring/file"
+);
+if (response.ok) {
+  console.log("File deleted !");
+}
+```
+
+## Writing a file
+
+There are two approaches to writing files:
+
+1. you know exactly at which IRI your file should be saved (potentially overwriting any data that sat there previously)
+2. you know what resource should be the parent of your file, like saving it into a folder.
```

Comment:
```suggestion
2. you know what [Container](../glossary#container) should be the parent of your file, like saving it into a folder.
```

On the one hand we might not want to require knowledge of what a Container is here, on the other hand using Container and Resource interchangeably might _also_ be confusing :/

Reply:
I think you're right, even if requiring readers to know about containers is not ideal, using Resource and Container interchangeably is more confusing.